src/shogun/lib/any.h

```diff
@@ -0,0 +1,291 @@
+    {
+    public:
+        /** Puts provided value pointed by v (untyped to be generic) to storage.
+         * @param storage pointer to a pointer to storage
+         * @param v pointer to value
+         */
+        virtual void set(void** storage, const void* v) const = 0;
+        
+        /** Clears storage.
+         * @param storage pointer to a pointer to storage
+         */
+        virtual void clear(void** storage) const = 0;
+        
+        /** Returns type-name as string.
+         * @return name of type class
+         */
+        virtual std::string type() const = 0;
+        
+        /** Compares type.
+         * @param ti type information
+         * @return true if type matches
+         */
+        virtual bool matches(const std::type_info& ti) const = 0;
+
+        /** Compares two storages.
+         * @param storage pointer to a pointer to storage
+         * @param other_storage pointer to a pointer to another storage
+         * @return true if both storages have same value
+         */
+        virtual bool equals(void** storage, void** other_storage) const = 0;
+    };
+    
+    /** @brief This is one concrete implementation of policy that
+     * uses void pointers to store values.
+     */
+    template <typename T>
+    class PointerValueAnyPolicy : public BaseAnyPolicy
+    {
+    public:
+        /** Puts provided value pointed by v (untyped to be generic) to storage.
+         * @param storage pointer to a pointer to storage
+         * @param v pointer to value
+         */
+        virtual void set(void** storage, const void* v) const
+        {
+            *(storage) = new T(*reinterpret_cast<T const*>(v));
+        }
+        
+        /** Clears storage.
+         * @param storage pointer to a pointer to storage
+         */
+        virtual void clear(void** storage) const
+        {
+            delete reinterpret_cast<T*>(*storage);
+        }
+        
+        /** Returns type-name as string.
+         * @return name of type class
+         */
+        virtual std::string type() const
+        {
+            return demangledType<T>();
+        }
+        
+        /** Compares type.
+         * @param ti type information
+         * @return true if type matches
+         */
+        virtual bool matches(const std::type_info& ti) const
+        {
+            return typeid(T) == ti;
+        }
+
+        /** Compares two storages.
+         * @param storage pointer to a pointer to storage
+         * @param other_storage pointer to a pointer to another storage
+         * @return true if both storages have same value
+         */
+        bool equals(void** storage, void** other_storage) const
+        {
+            int typed_storage = *(reinterpret_cast<int*>(*storage));
+            int typed_other_storage = *(reinterpret_cast<int*>(*other_storage));
+            return typed_storage == typed_other_storage;
+        }
+    };
+
+    /** @brief Allows to store objects of arbitrary types. */
```

Comment:
More infos please as any is at such core of this new framework.
Reference the classes it is used in